cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHttpService.java

```diff
@@ -97,6 +100,12 @@ public InetSocketAddress getSocketAddress() {
     janitorService.startAndWait();
     streamCoordinator.setHandlerDiscoverable(discoverable);
     streamCoordinator.startAndWait();
+
+    // In the in-memory implementations, streamCoordinator and streamCoordinatorLeader are the same implementation,
+    // hence we don't want to start them twice.
+    if (!streamCoordinatorClient.isRunning()) {
```

Comment:
I think it's ok to just call start. Trying to start a already started guava service has no effect.